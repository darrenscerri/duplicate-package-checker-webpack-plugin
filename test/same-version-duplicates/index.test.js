var webpack = require("webpack");
var assert = require("assert");
var MakeConfig = require("./make.webpack.config");

describe("Duplicate submodule dependency tree", function() {
  it("should output warnings if ignoreSameVersionDuplicates == false", function(done) {
    webpack(
      MakeConfig({
        ignoreSameVersionDuplicates: false
      }),
      function(err, stats) {
        assert(stats.compilation.warnings.length === 1);
        expect(stats.compilation.warnings[0].message).toMatchSnapshot();
        done();
      }
    );
  });

  it("should not output warnings if ignoreSameVersionDuplicates == true", function(done) {
    webpack(MakeConfig(), function(err, stats) {
      assert(stats.compilation.warnings.length === 0);
      done();
    });
  });
});
