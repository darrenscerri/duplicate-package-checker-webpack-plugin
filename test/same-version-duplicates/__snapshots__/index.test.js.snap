// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Duplicate submodule dependency tree should output warnings if ignoreSameVersionDuplicates == false 1`] = `
"a
[0m  Multiple versions of [0m[32m[1ma[22m[39m[37m found:[39m
[37m[39m    [32m[1m2.0.0[22m[39m [37m[1m./~/c/~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/b/~/a[22m[39m
    [32m[1m2.0.0[22m[39m [37m[1m./~/a[22m[39m

[37m[1mCheck how you can resolve duplicate packages: [22m[39m
https://github.com/darrenscerri/duplicate-package-checker-webpack-plugin#resolving-duplicate-packages-in-your-bundle
"
`;
